src/frontend/src/lib/components/convert/ConvertInputAmount.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { IconClose } from '@dfinity/gix-components';
+	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import InputCurrency from '$lib/components/ui/InputCurrency.svelte';
+	import type { Token } from '$lib/types/token';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let token: Token;
+	export let amount: number | undefined = undefined;
+	export let disabled: boolean | undefined = undefined;
+	export let customValidate: (userAmount: BigNumber | undefined) => void = () => {};
+	export let error: Error | undefined = undefined;
+	export let placeholder = '0.00';
+
+	const onReset = () => {
+		amount = undefined;
+	};
+
+	const validate = () => {
+		const parsedValue = !invalidAmount(amount)
+			? parseToken({
+					value: `${amount}`,
+					unitName: token.decimals
+				})
+			: undefined;
+
+		customValidate(parsedValue);
+	};
+
+	const debounceValidate = debounce(validate, 300);
+
+	$: amount, token.decimals, debounceValidate();
+</script>
+
+<div class="convert-input-amount text-3xl font-bold" class:error={nonNullish(error)}>
+	<InputCurrency
+		name="convert-amount"
+		bind:value={amount}
+		decimals={token.decimals}
+		{placeholder}
+		{disabled}
+	>
+		<svelte:fragment slot="inner-end">
+			{#if nonNullish(amount) && !disabled}
+				<button
+					on:click|preventDefault={onReset}
+					class={nonNullish(error) ? 'text-error' : 'text-aurometalsaurus'}
+					class:hidden={isNullish(amount) || disabled}
+				>
+					<IconClose />
+				</button>
+			{/if}
+		</svelte:fragment>
+	</InputCurrency>
+</div>
+
+<style lang="postcss">
+	:global(.convert-input-amount) {
```

Comment:
is there a way without using global?

Reply:
I did try everything I could come up with - unfortunately, without `global` the rules are considered as "local" and therefore are not applied to the children components.

Reply:
And there is no CSS4 variables already available for those? Maybe in another PR, we should then extend the underneath components with variables and then get rid of this global. That's prone to maintainance issue. We gonna need an E2E test with screenshot for it.

Reply:
For some rules, I used the variables to update the styling. But, for example, for `error` state I had to rewrite the whole property. And yeah, totally agree that we need to improve this part as well as have e2e screenshot for such a case.